src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -425,6 +441,38 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+/// <summary>
+/// Set the CustomHttpHeaders property from the custom http headers string.
+/// </summary>
+/// <param name="stringValue">The value to set.</param>
+public void ValidateAndSetCustomHttpHeadersFromString(string headersValue) 
+{
+    if (String.IsNullOrWhiteSpace(headersValue))
+    {
+        return;
+    }
+
+    this.CustomHttpHeaders = new List<string>();
+
+    string[] headerElements = headersValue.Split(new string[] { Environment.NewLine }, StringSplitOptions.None);

```

Comment:
Question, is it possible for the headersValue to contain a newline somewhere? start, end, middle? If so, then ```StringSplitOptions.RemoveEmptyEntries``` might be better so that the tests below don't throw for an empty string. If having empty lines is not desired, then it works as is.

Reply:
Changes to RemoveEmptyEntries